apps/api/src/services/competition.service.ts

```diff
@@ -2265,6 +2266,107 @@ export class CompetitionService {
     }
   }
 
+  /**
+   * Check and automatically start competitions that have reached their start date
+   * Conditions:
+   * - No other competition is currently active
+   * - Competition is not in sandbox mode
+   * - Competition has at least one registered agent
+   * - Process competitions by earliest startDate first
+   */
+  async processCompetitionStartDateChecks(): Promise<void> {
+    try {
+      // Do not start anything if there's already an active competition
+      const active = await findActive();
+      if (active) {
+        serviceLogger.debug(
+          `[CompetitionManager] Active competition found (${active.id}). Skipping auto-start checks`,
+        );
+        return;
+      }
+
+      const competitionsToStart = await findCompetitionsNeedingStarting();
+      if (competitionsToStart.length === 0) {
+        serviceLogger.debug(
+          "[CompetitionManager] No competitions ready to start",
+        );
+        return;
+      }
+
+      // We only support running one competition at a time, so we will not start any competitions
+      // if we find more than one. Note: This should not happen if competitions are created with
+      // the correct start dates; it's defensive.
+      if (competitionsToStart.length > 1) {
```

Comment:
i figure we want to be extra careful, and if there is more than one "ready" competition (in a `pending` state), that's probably due to human error with dates. thus, we just skip all comps and don't start any of them.